libnd4j/include/config.h.in

```diff
@@ -19,4 +19,12 @@
 
 #cmakedefine DEFAULT_ENGINE @DEFAULT_ENGINE@
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+const char* buildInfo();
```

Comment:
This doesn't belong in config.h.

Reply:
@saudet  the goal here is to be able to print how libnd4j was built.
If we want to expose/print that, where would you recommend it?

Reply:
I'll move those to build_info.h. 
If it is desired I'm thinking of renaming both to version.h and adding internal libnd4j version too. that returns a structure of {major;minor;patch}
actually, I opened this Pr for discussions.